mod-source-record-storage-server/src/main/java/org/folio/dao/QueryExecutorInterceptor.java

```diff
@@ -0,0 +1,148 @@
+import net.bytebuddy.implementation.SuperMethodCall;
+import net.bytebuddy.matcher.ElementMatchers;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.nio.channels.ClosedChannelException;
+import java.util.concurrent.Callable;
+import java.util.function.Supplier;
+
+/**
+ * This class provides an interceptor for the `transaction` and `query` methods of
+ * `ReactiveClassicGenericQueryExecutor`. It includes functionality for retrying the
+ * operations based on a configurable number of retries and delay, specifically for
+ * exceptions like `NoStackTraceThrowable` or `ClosedChannelException`.
+ */
+public class QueryExecutorInterceptor {
+  private static final Logger LOGGER = LogManager.getLogger();
+  private static int numRetries = 1;
+  private static long retryDelay = 1000;
+
+  private QueryExecutorInterceptor(){}
+
+  /**
+   * Sets the number of retries for the transaction and query methods.
+   *
+   * @param numberOfRetries the number of retries to set
+   */
+  public static void setNumberOfRetries(int numberOfRetries) {
+    numRetries = numberOfRetries;
+  }
+
+  /**
+   * Sets the delay time between retries in milliseconds.
+   *
+   * @param delay the delay time to set in milliseconds
+   */
+  public static void setRetryDelay(long delay) {
+    if (delay < 0) throw new IllegalArgumentException("delay cannot be less than zero");
+    retryDelay = delay;
+  }
+
+  /**
+   * Generates a subclass of `ReactiveClassicGenericQueryExecutor` with interceptors for the
+   * transaction and query methods.
+   *
+   * @return the generated subclass
+   */
+  public static Class<? extends ReactiveClassicGenericQueryExecutor> generateClass() {
+    return new ByteBuddy()
+      .subclass(ReactiveClassicGenericQueryExecutor.class)
+      .constructor(ElementMatchers.any()) // Match all constructors
+      .intercept(SuperMethodCall.INSTANCE) // Call the original constructor
+      .method(ElementMatchers.named("transaction")) // For transaction method
+      .intercept(MethodDelegation.to(QueryExecutorInterceptor.class))
+      .method(ElementMatchers.named("query")) // For query method
+      .intercept(MethodDelegation.to(QueryExecutorInterceptor.class))
+      .make()
+      .load(QueryExecutorInterceptor.class.getClassLoader())
+      .getLoaded();
+  }
+
+  /**
+   * Interceptor for the query method, with retry functionality.
+   *
+   * @param superCall the original method call
+   * @return the result of the query operation
+   */
+  @SuppressWarnings("unused")
+  public static <U> Future<U> query(
+    @net.bytebuddy.implementation.bind.annotation.SuperCall Callable<Future<U>> superCall
+  ) {
+    LOGGER.trace("query method of ReactiveClassicGenericQueryExecutor proxied");
+    return retryOf(() -> {
+      try {
+        return superCall.call();
+      } catch (Throwable e) {
+        LOGGER.error("Something happened while attempting to make proxied call for query method", e);
```

Comment:
@okolawole-ebsco As far as I remember for our current approach we set all errors as WARN-level for better errors-handling process on the envs. Or do you think this is an exceptional case for the whole module here? (and other places in this PR)

Reply:
Yeah I think it is an exceptional case. I don't expect this to happen often

Reply:
@okolawole-ebsco  Thank you!